docs/DeveloperGuide.md

```diff
@@ -1,13 +1,269 @@
-# Developer Guide
+# Developer Guide for MangaTantou
 
-## Acknowledgements
+# Acknowledgements
+MangaTantou's structure has been greatly inspired by the team developers' respective iPs as listed below:
+1. [Donovan](https://github.com/xenthm/ip)
+2. [Sarah](https://github.com/sarahchow03/ip)
+3. [Ian](https://github.com/iaso1774/ip)
+4. [Andy](https://github.com/averageandyyy/ip)
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+The following third-party libraries were used:
+- [Gson 2.11.0](https://github.com/google/gson): A Java library for converting Java objects to JSON and vice versa, licensed under Apache License 2.0.
 
-## Design & implementation
+Additionally, the following resources/websites were heavily used (they are amazing): 
+- [RegExr](https://regexr.com/): RegExr is an online tool to learn, build, & test Regular Expressions (RegEx / RegExp).
+- [Regex Vis](https://regex-vis.com/): Regex visualizer and editor. 
+- [RegEx101 ](https://regex101.com/): Regex visualizer,editor, and debugger.  
+- [Ashley's PlantUML Doc](https://plantuml-documentation.readthedocs.io/): Documentation about how to use the commands, keywords, options, and other information needed to produce diagrams with PlantUML. 
 
+# Design & implementation
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+## Overall Architecture
+### Representing Data in MangaTantou
+![AuthorListClass.png](uml/puml/AuthorListClass/AuthorListClass.png)
+The above UML class diagram shows the overall structure of author and manga data an editor using MangaTantou would be interested in. 
+
+> **_NOTE:_** There is circular reference (bidirectional navigability) between `Author` and `Manga` through `MangaList`. 
+
+### Commands 
+![Command Inheritance](/docs/uml/puml/Command/Command.png)
```

Comment:
Perhaps the Command.png can be placed in the images folder! As currently it could not load in the DG 😿 

![image](https://github.com/user-attachments/assets/63e3f3d7-23cd-4959-b36b-0c76ccf2cbf6)

Reply:
Ah, careless mistake! Thanks for helping us point it out early!